datastructures/King.cpp

```diff
@@ -0,0 +1,66 @@
+#include "King.h"
+
+King::King(const Position &initialPosition, const Color &initialPlayer) : Piece(initialPosition, initialPlayer) {}
+
+std::vector<Position> King::possibleMoves() {
+    std::vector<Position> moves;
+    Position possiblePositions;
+
+    // Handles left side moves
+    if (currentPosition.column - 1 >= 'A') {
```

Comment:
you can put `currentPosition.column - 1` in a variable instead of computing it twice.